hal/src/sercom/v2/uart/reg.rs

```diff
@@ -0,0 +1,221 @@
+//! Register-level access to UART configuration
+
+use super::{BaudMode, Parity, StopBits};
+
+use crate::sercom::v2::*;
+use crate::time::Hertz;
+
+pub(super) struct Registers<S: Sercom> {
+    pub(super) sercom: S,
+}
+
+impl<S: Sercom> Registers<S> {
+    /// Helper function to access the given `SERCOM`
+    pub(crate) fn sercom(&self) -> &S {
+        &self.sercom
+    }
+
+    /// Helper function to access the underlying `USART` from the given `SERCOM`
+    #[cfg(any(feature = "samd11", feature = "samd21"))]
+    pub(crate) fn usart(&self) -> &crate::target_device::sercom0::USART {
+        self.sercom().usart()
```

Comment:
You can just access the `sercom` field directly here. There's no need for the `.sercom()` method anymore.